src/nisarqa/processing/insar/quiver_plots.py

```diff
@@ -352,26 +580,99 @@ def truncate_colormap(
+                az_offsets=az_offsets_at_arrow_tails,
+                rg_offsets=rg_offsets_at_arrow_tails,
+                geo_grid=nisarqa.GeoGrid(
+                    epsg=geo_grid.epsg,
+                    x_coordinates=geo_grid.x_coordinates[::arrow_stride],
+                    x_spacing=geo_grid.x_spacing * arrow_stride,
+                    y_coordinates=geo_grid.y_coordinates[::arrow_stride],
+                    y_spacing=geo_grid.y_spacing * arrow_stride,
+                ),
+                projection_params=quiver_projection_params,
+            )
+        )
+
+    # According to the documentation for `matplotlib.pyplot.quiver()`,
+    # when you pass angles='xy', the arrow offsets should be in the same
+    # coordinates as the arrow tail locations, i.e. the arrows will point
+    # from (x, y) to (x+u, y+v).
+    # Here, we are storing the arrow tail locations are in pixels,
+    # while the x and y offsets are in units of meters. So, what we
+    # actually want to plot are the arrow offsets in pixels (i.e. the
+    # same units as the arrow tail locations).
+    # So let's divide the offsets by the pixel spacing.
+
+    # Note: The y-coordinate posting of the coordinate grid of NISAR L2
+    # products is negative (the positive y-axis points up in the plot).
+    # That is, a postive-valued offset in meters will be a negative-valued
+    # offset in pixels. So, when you divide the offsets by the pixel
+    # spacing, it should flip the sign of the y component of the offsets.
+
+    # Conversely, the y-coordinate posting of NISAR L1 products is
+    # positive (the positive y-axis points down in the plot).
+    # So dividing by the pixel spacing should not flip the sign of the
+    # y component of the offsets when making quiver plots for L1 products.
+    arrow_heads_y_offsets_in_pixels = arrow_heads_y_offsets / y_posting
+    arrow_heads_x_offsets_in_pixels = arrow_heads_x_offsets / x_posting
+
     # Add the quiver arrows to the plot.
     # Multiply the start and end points for each arrow by the decimation factor;
     # this is to ensure that each arrow is placed on the correct pixel on
     # the full-resolution `total_offset` background image.
     ax.quiver(
-        # starting x coordinate for each arrow
-        X * arrow_stride,
-        # starting y coordinate for each arrow
-        Y * arrow_stride,
-        # ending x direction component of for each arrow vector
-        arrow_x * arrow_stride,
-        # ending y direction component of for each arrow vector
-        arrow_y * arrow_stride,
+        # starting pixel index in x direction for each arrow tail
+        arrow_tails_x_indices,
+        # starting pixel index in y direction for each arrow tail
+        arrow_tails_y_indices,
+        # x offset for each arrow's tip
+        arrow_heads_x_offsets_in_pixels * arrow_stride,
+        # y offset for each arrow's tip
+        arrow_heads_y_offsets_in_pixels * arrow_stride,
```

Comment:
Scaling the offsets by a scalar factor won't affect the direction of the arrows -- only the magnitude. The magnitude here is basically arbitrary because Matplotlib will auto-scale them, but I don't think scaling by the stride makes intuitive sense anyway, so I'd suggest removing that. It shouldn't have any effect on the plotted output.

```suggestion
        arrow_heads_x_offsets_in_pixels,
        # y offset for each arrow's tip
        arrow_heads_y_offsets_in_pixels,
```